nimble/controller/src/ble_ll_adv.c

```diff
@@ -2646,6 +2677,7 @@ ble_ll_adv_sm_start_periodic(struct ble_ll_adv_sm *advsm)
      * train is enabled.
      */
     ble_ll_adv_update_did(advsm);
+    ble_ll_adv_update_did_periodic(advsm);
```

Comment:
there's no need to update DID for periodic advertising because periodic advertising data doesn't change here